Long token symbols layout updates

fairlighteth opened the pull request:
# Summary

- This addresses https://github.com/cowprotocol/cowswap/issues/1852 by iterating on https://github.com/cowprotocol/cowswap/pull/1854 
- I think it's important to show the full symbol name at all times. I understand it becomes less useful beyond XX amount of characters. But in this PR I made sure our layout can fully support longer token names without kicking in the text ellipsis too soon.
- In this PR I've set `const DEFAULT_MAX_SYMBOL_LENGTH = 40` to have some kind of 'hard cap' for really long tokens. Technically in terms of layout it wouldn't be necessary (we can support it), although I think the argument of it being less useful might be more compelling. So I think we should review/fine tune this max symbol length.
- This applies to both the SWAP and LIMIT ORDER UI.


https://user-images.githubusercontent.com/31534717/212709802-38e8de24-352b-4296-b471-447accb5bb3d.mov

https://user-images.githubusercontent.com/31534717/212709837-0f60c518-9cf5-49ca-8fe6-523bdcaf6929.mov

anxolin reviewed (approved):
Yes, you are right its best to show the full name. 

I just thought in some places can be hard, and same as in Etherscan they show the ellipsos and tooltip, we could do something similar. 

However, if we can show them all, its better. Some complex token derivatives might have super important info in their name

fairlighteth commented:
Provided a fix for point 1. @alfetopito please see if this makes sense to you (https://github.com/cowprotocol/cowswap/pull/1881/commits/09cd0ae50fb7d3217f61246b415864e7268cb25d)

fairlighteth commented:
Fixed point 2 and 3:

https://user-images.githubusercontent.com/31534717/212887672-d26423be-3277-4b0b-8e01-b985e49758e5.mov

Also I noticed that the limit price rates are abbreviated on fetching a new price. I wonder if this is desired behavior? E.g. if I happen to want a limit price of `0.0000000000000001` it abbreviates it to `<0.000001` which in the context of setting a precise limit order might not be desired behavior? Example:

https://user-images.githubusercontent.com/31534717/212887940-c4d7fda6-b6f6-483e-b87c-3c5351468a50.mov

fairlighteth commented:
As for point 4, I went for a more complete iteration, by prioritizing the token symbol, rather than the name. Comparing different portfolio web UI's I noticed a similar pattern and found it easier to scan tokens by SYMBOL. Furthermore I made sure that token symbols/names are rendered in a more readable way:

**BEFORE:**
<img width="1122" alt="Screenshot 2023-01-17 at 11 41 34" src="https://user-images.githubusercontent.com/31534717/212890654-c15b88f7-7371-4a1c-9fb5-f1213532146c.png">

**AFTER:**

<img width="1136" alt="Screenshot 2023-01-17 at 11 41 16" src="https://user-images.githubusercontent.com/31534717/212890733-adbb2d87-4445-48f5-8830-53184d394ddd.png">
<img width="1154" alt="Screenshot 2023-01-17 at 11 41 05" src="https://user-images.githubusercontent.com/31534717/212890739-25de415f-e5d5-4dab-985e-01de0f64156d.png">

https://user-images.githubusercontent.com/31534717/212890764-b6873cc7-84f2-4d40-9b9e-efcd9ddbc3b6.mov

elena-zh commented:
> Also I noticed that the limit price rates are abbreviated on fetching a new price. I wonder if this is desired behavior? E.g. if I happen to want a limit price of `0.0000000000000001` it abbreviates it to `<0.000001` which in the context of setting a precise limit order might not be desired behavior?

Hey @fairlighteth , I think that this is something that should be addressed when fixing these issues (included into the tech debt):  https://cowservices.slack.com/archives/C0361CDG8GP/p1671548637214759

elena-zh commented:
Hey @fairlighteth , great changes!

However, please, don't kill me, but seems that I have found several new places with a long_token_name_issues: 
1. Confirm order modal: 
![confirm modal](https://user-images.githubusercontent.com/70885163/212894785-7b5eb36e-52d6-431a-9949-d363e5e5d666.png)
2. Limit price label on the Account modal (mobile view)
![lim price](https://user-images.githubusercontent.com/70885163/212895043-ead6e99b-6e8f-4eee-bed4-fa46ac0812b4.png)

3. Limit price label on the Your orders table
![limit price](https://user-images.githubusercontent.com/70885163/212895086-eaa2f306-c6d5-4c6f-aeb8-cc4a3005cd5d.png)

4. Swap form
![some places](https://user-images.githubusercontent.com/70885163/212895181-4dbff7c5-0bd1-4d70-9bab-38fedfae03c1.png)
5. As for the tokens table, I would disable titles on hover, as all names are clearly visible now
![remove on hover](https://user-images.githubusercontent.com/70885163/212895299-69171ebd-e3b4-4fcb-973c-e2a275f981ea.png)

Thank you!

fairlighteth commented:
@elena-zh Thanks for reporting the issues! I think I've addressed them all now. I made a refactor to **trim** all symbols to 12 characters by default (`const DEFAULT_MAX_SYMBOL_LENGTH = 12`).

Then added an optional prop `maxLength` to the function `formatSymbol` and component `TokenSymbol` which allows you to pass a custom maxLength. The rationale here is that I think we should set a 'common' max symbol length that is reasonable.

In some instances I think it's important to show the full token symbol. Mainly for visual confirmation that you as a user, are dealing with the right token (symbol). For instance in the Token Selector, I think it makes sense to show a bigger length. I there applied a `maxLength` of `40`. 

Also I applied the function `formatSymbol` on other places where the currency symbol is used (except for wrapped/native token mentions).

elena-zh commented:
Hey @fairlighteth , thank you! 
I still have some nitpicks here (seems to be an endless PR): 
Favourite token in a mobile view: 
![image](https://user-images.githubusercontent.com/70885163/213144194-e9166fbb-de22-4d59-bad4-524320086a0c.png)

Balances in a mobile view:
![image](https://user-images.githubusercontent.com/70885163/213144288-c0e60595-8c39-4851-bfac-b59d25494455.png)

Order confirm modal: I think we should leave a bit more space for an amount. WDYT?
![not to crop amounts](https://user-images.githubusercontent.com/70885163/213144522-53eb097b-19f3-41e0-9c8c-2dba9b00eda5.png)

And still I vote for removing title on hover a token name/symbol on Tokens page
![remove on hover](https://user-images.githubusercontent.com/70885163/213144954-a4f2932a-88cc-467c-a99f-77ae7129be1c.png)

fairlighteth commented:
@elena-zh Thanks for reviewing again. Addressed your points and also including:

- Add to MetaMask also uses the `formatSymbol` (Add TOKEN to Metamask)
- Refactored `formatSymbol` to just use the `length` prop and not the extra `maxLength` prop I introduced earlier.

One way to quickly see where the `formatSymbol` is being used in the UI, is to edit file `src/cow-react/utils/format.ts` and set the maxLength to 1:
> const DEFAULT_MAX_SYMBOL_LENGTH = 1

Anywhere where you then see more than 1 character for the token symbol means:
- It has a specific `length` prop override (on purpose)
- It is not using the `formatSymbol` 

https://user-images.githubusercontent.com/31534717/213217931-7b5383a7-7a8a-4106-a66b-a9554196e9c3.mov


https://user-images.githubusercontent.com/31534717/213217957-dee9c403-8fa6-4c42-a391-65954adbaba7.mov

elena-zh reviewed (approved):
LGTM now!

elena-zh commented:
The issue for removing titles on hover a token name is #1106

alfetopito commented on src/cow-react/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx:
```diff
@@ -137,7 +138,7 @@ export function CurrencyInputPanel(props: CurrencyInputPanelProps) {
             {balance && !disabled && (
               <>
-                <styledEl.BalanceText title={balance.toExact() + ' ' + currency?.symbol}>
+                <styledEl.BalanceText title={balance.toExact() + ' ' + formatSymbol(currency?.symbol)}>
```
Are we sure we want to trim token symbols in the title?
It's hidden and wouldn't impact the UI. Seeing the whole thing there would be beneficial IMO.

fairlighteth replied:
Actually we shouldn't and this was an oversight.
